source/distributed/fully_distributed_tria.cc

```diff
@@ -184,7 +184,9 @@ namespace parallel
             }
 
           // 4b) set actual (level_)subdomain_ids
-          for (unsigned int level = 0; level < cell_infos.size(); ++level)
+          for (unsigned int level = 0;
+               level < cell_infos.size() && !cell_infos[level].empty();
+               ++level)
```

Comment:
I haven't tried to understand what this does, but did you really mean to terminate the loop if the `cell_infos[level]` is empty? Or did you just want to skip that level?